rts/motoko-rts/src/gc/mark_compact/bitmap.rs

```diff
@@ -90,36 +151,36 @@ pub const BITMAP_ITER_END: u32 = 1024 * 1024 * 1024;
 impl BitmapIter {
     /// Returns the next bit, or `BITMAP_ITER_END` if there are no more bits set.
     pub fn next(&mut self) -> u32 {
-        debug_assert!(self.current_word_idx <= self.size);
+        debug_assert!(self.current_bit_idx <= self.size);
 
         // Outer loop iterates 64-bit words
         loop {
-            if self.current_word == 0 && self.current_word_idx == self.size {
+            if self.current_word == 0 && self.current_bit_idx == self.size {
```

Comment:
I have the feeling that this can still be tweaked, maybe just `self.current_bit_idx >= self.size`?

Reply:
this is better now